script/phase-1b/01-FarmingDeploy.s.sol

```diff
@@ -0,0 +1,48 @@
+// SPDX-FileCopyrightText: © 2023 Dai Foundation <www.daifoundation.org>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+pragma solidity ^0.8.16;
+
+import {Script} from "forge-std/Script.sol";
+import {ScriptTools} from "dss-test/ScriptTools.sol";
+
+import {Reader} from "../helpers/Reader.sol";
+import {StakingRewardsDeploy, StakingRewardsDeployParams} from "../dependencies/StakingRewardsDeploy.sol";
+
+contract Phase1b_FarmingDeployScript is Script {
+    string internal constant NAME = "phase-1b/farming-deploy";
+
+    function run() external {
+        Reader reader = new Reader(ScriptTools.loadConfig());
+
+        address admin = reader.envOrReadAddress("FOUNDRY_ADMIN", ".admin");
+        address nst = reader.envOrReadAddress("FOUNDRY_NST", ".nst");
+        address rewards = reader.readAddressOptional(".rewards");
+
+        vm.startBroadcast();
+
+        if (rewards == address(0)) {
+            rewards = StakingRewardsDeploy.deploy(
+                StakingRewardsDeployParams({owner: admin, stakingToken: nst, rewardsToken: address(0)})
+            );
+        }
```

Comment:
```suggestion
        address rewards = StakingRewardsDeploy.deploy(
            StakingRewardsDeployParams({owner: admin, stakingToken: nst, rewardsToken: address(0)})
        );
```

Reply:
Hmmm, I was confused by that. I removed it initially but i understood that the check was in case `rewards` was already deployed a new deployment was skipped, like in the other scripts in `phase-0 `fro example.

Reply:
I added this sort of conditional path because in the past, Foundry would just bail the script whenever a transaction fail, leaving some dangling contracts that were not properly wired together. To not waste gas, the deployer could pick whatever had been deployed, set the addresses in the config and the script would run from where it stopped.

Since then, the transaction sending mechanism for Foundry seems to have improved, and also we realized that using the `--slow` flag made the problems we had before go away. We should probably refactor the other scripts, but they have been audited already.